src/RA_StringUtils.h

```diff
@@ -190,14 +187,30 @@ class StringBuilder
         m_vPending.emplace_back(std::move(pPending));
     }
 
+    void Append(const std::string&& arg)
```

Comment:
I think you may have misunderstood something, but that's OK.
rvalue references that are const are useless (as well as undefined).
The purpose of std::move is to eliminate copying, the right side has to be in a default state. The left side steals the contents of the right side. 

Example:
```
arg is const&&: auto str = std::move(arg); // arg is unchanged, str is empty
arg is &&: auto str = std::move(arg); // str = arg, arg is empty.
```
If you wanted to use a reference wrapper, this wouldn't even compile:
```
template <class T>
void ref(const T&&) = delete;

template <class T>
void cref(const T&&) = delete;
```